README.md

```diff
@@ -0,0 +1,114 @@
+# xtm - a library for cross-thread messaging with event loop intergation
+
+The library provides the following facilities:
+
+# xtm_queue
+
+Opaque struct, that represent one-directional, one-reader-one-writer queue
+implementation with eventloop integration ability.
+
+## xtm_queue_create
+
+Allocation function, used to create instance of struct xtm_queue and return
+its pointer or NULL in case of error. Accepts the queue size as input, which
+must be a power of two.
+
+## xtm_queue_delete
+
+Deallocation function, used to free queue and close its internal fds.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+close(2), since it implies close of fds)
+
+## xtm_msg_notify
+
+Function for queue consumer notification.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+write(2), since it implies write to internal fd)
+
+## xtm_msg_probe
+
+Returns 0 if queue has space. Otherwise -1 with errno set to ENOBUFS.
+
+## xtm_msg_count
+
+Function returns current message count in queue.
+
+## xtm_fun_dispatch
+
+Function puts message containing the function and its argument to the queue.
+If delayed == 0 notifies consumer.
+Returns 0 on success. Otherwise -1 with errno set appropriately (as in
+write(2), since it implies write to internal fd + ENOBUFS in case there
+is no space in queue)
+
+## xtm_fd
+
+Returns file descriptor, that should be watched to become readable. When it
+became readable, consumer should call one of the consumer functions:
+xtm_msg_recv or xtm_fun_invoke.
+
+## xtm_fun_invoke
+
+Function retrieves messages from the queue and calls functions contained
```

Comment:
In case you use xtm_msg_send with xtm_fun_invoke, you catch SIGSEGV, because you can't call msg. We can add check (for example, flag inside each message). But it seems to me that this should remain on the user's conscience, since such a check can negatively affect performance.

Reply:
I am fine without a check if this is intended. But it wouldn't be needed if a user would be able to pass any 16 byte blocks. Then there wouldn't be a place to crash.